Sources/Quick/AllExamplesBuilder.swift

```diff
@@ -0,0 +1,68 @@
+#if !SWIFT_PACKAGE
+
+import Foundation
+import XCTest
+
+/// A dummy protocol for calling the internal `+[QuickSpec buildExamplesIfNeeded]` method
+/// which is defined in Objective-C from Swift.
+@objc private protocol _QuickSpecInternal {
+    static func buildExamplesIfNeeded()
+}
+
+internal final class AllExamplesBuilder {
+    private var didBuildAllExamples = false
+
+    /// Calls `buildExamplesIfNeeded` on each `QuickSpec` subclass
+    func buildAllExamplesIfNeeded() {
+        guard !didBuildAllExamples else { return }
+        didBuildAllExamples = true
+
+        QuickSpec.enumerateSubclasses { specClass in
+            // This relies on `_QuickSpecInternal`.
+            (specClass as AnyClass).buildExamplesIfNeeded()
+        }
+    }
+}
+
+// swiftlint:disable:next todo
+// TODO: Unify this with QuickConfiguration's equivalent
```

Comment:
I think we should address this before merging. 

But I don't have a ton of context here.